### DISCUSSION TITLE
Implementing similar Everywhere Actions using Model Viewer in QuickLook & Scene Viewer

### DISCUSSION THREAD

**Discussion by StillOnMyWay** [google/model-viewer#4929] [Q&A]
Thanks for all the work that has been done on Model Viewer, following for over a year now.

Needle.tools has implemented an example of what they describe as Everywhere Actions, with some very exciting demos in AR Mode. To get something like the musical example they share first, is this a GLB question (is this interactivity in the file itself) or is this something that can't be accomplished with Model Viewer if you want to use Quick Look for iOS or Scene Viewer for Android. Reference - https://engine.needle.tools/docs/everywhere-actions.html#how-do-i-use-everywhere-actions

I'm aware that I can do this in the AR mode by handling the touch/click of the material with Javascript and responding, but really wondering how did they accomplish this, and can I as well with Model Viewer?

**Comment by elalish** [comment]
Thank you - yes there is a new glTF extension spec under development for putting these kinds of interactive behavior graphs directly into the GLB. MV does not yet support them, but as soon as Three.js builds support, we'll pull it in. As for AR, as usual we'll get it for free in our default WebXR mode on Android. I don't think we'll support these features in USDZ autogeneration, but I believe the purpose of @hybridherbst's Everywhere Actions is that their tools can export a USDZ that you can use with AR QuickLook in MV by setting `ios-src`. Hard to say when Scene Viewer will get support, but possibly before MV does.
